src/components/SearchBar.tsx

```diff
@@ -55,17 +56,29 @@ const SearchBar: React.FC<SearchBarProps> = (props) => {
   const history = useHistory();
 
   const onChangeHandler = (label: string | null) => {
-    const item = _.find<SearchItem>(data, { label: label ?? "" });
+    const item = find<SearchItem>(data, { label: label ?? "" });
     if (!item) {
       history.push(`/search?label=${label}`);
     } else if (item.isWord) {
       history.push(`/disambiguation?label=${label}`);
     } else {
       const prop = item.items[0];
-      history.push("/term", prop);
+      history.push(generateTermRoute(prop));
     }
   };
 
+  const handleChange = (event: ChangeEvent<{}>, newInputValue: string) => {
+    setInputValue(newInputValue);
+  };
+
+  const debouncedChangeHandler = useMemo(() => debounce(handleChange, 300), []);
+
+  useEffect(() => {
+    return () => {
+      debouncedChangeHandler.cancel();
+    };
+  });
```

Comment:
Seems like dependencies are missing in this hook. I assume you want to cancel the handler upon component unmount, in which case you need to add `[]` as dependencies. Because if you don't then I believe that the cancel will be fired multiple times - based on how many times the component was previously rendered.